mantidproject/mantid
# Keren function: fix and test with real data

The [Keren](http://docs.mantidproject.org/nightly/fitfunctions/Keren.html) fit function was added new in release 3.7 so that scientists could try it out when data became available.

Now that data is available, fixed a problem that arose from these tests where the field came out with the wrong units. Mantid now gives results within the errors of those from WiMDA - added a system test to confirm this.

Also fixed some small errors in the documentation and added units to the parameter description strings.

**To test:**

The new system test compares Mantid's results to those obtained independently with another program (WiMDA). If you want to test manually as well, the data is available at `\\olympic\babylon5\Scratch\TomPerkins\Data\keren_data`.
- Load a data file into the Muon Analysis interface
- Go to "Data Analysis" tab
- Set up a function of `FlatBackground + Keren`
- Fit the data and compare the results to those in the `kerentest.dat` file, which are the WiMDA results. The results from Mantid should be close to those from WiMDA, though there may be some small differences.
  - In this file "a-baselin" corresponds to _f0.A0_ from the `FlatBackground` and "a-relaxin" is _f1.A_ from `Keren`. Both are x100 because WiMDA uses % asymmetry. "C1 Delta" and "C1 Fluct" are the _Delta_ and _Fluct_ parameters from Keren, and the Field was fixed in the WiMDA fits.

Fixes #16212.

[Release notes](https://github.com/mantidproject/mantid/blob/0c895db330eac562ebd25ea023b304aeae510e78/docs/source/release/v3.8.0/muon.rst#fit-functions) have been updated.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
